payment-handler/payment-instruments-manual.https.html

```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<link rel="help" href="https://w3c.github.io/payment-handler/">
+<title>Test for PaymentInstrument</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/service-workers/service-worker/resources/test-helpers.sub.js"></script>
+<script>
+
+function addInstruments(instruments) {
+  return Promise.all([
+    instruments.set(
+      'test_key1',
+      {
+        name: 'test_instrument1',
+        enabledMethods: ['test_pay']
+      }),
+
+    instruments.set(
+      'test_key2',
+      {
+        name: 'test_instrument2',
+        enabledMethods: ['basic-card']
+      })
+  ]);
+}
+
+promise_test(async test => {
+  const permission = PaymentManager.requestPermission();
+  assert_equals(permission, 'granted');
+
+  const registration = await service_worker_unregister_and_register(
+      test, 'resources/sw-payment-handler.js', 'resources/');
+  await wait_for_state(test, registration.installing, 'activated');
+
+  const instruments = registration.paymentManager.instruments;
+  await addInstruments(instruments);
+
+  const testInstrument1 = await instruments.get('test_key1');
+  assert_equals(testInstrument1.name, 'test_instrument1');
+  assert_array_equals(testInstrument1.enabledMethods, ['test_pay']);
+
+  const testInstrument2 = await instruments.get('test_key2');
+  assert_equals(testInstrument2.name, 'test_instrument2');
+  assert_array_equals(testInstrument2.enabledMethods, ['basic-card']);
+
+  // PaymentInstruments.get() should resolve `undefined` if no stored key
+  const undefinedInstrument = await instruments.get('undefined_key');
+  assert_equals(undefinedInstrument, undefined);
```

Comment:
Probably want a test that overrides an existing entry. 

Also, we probably want a test to check what equality of pulling the same instrument multiple times out of storage.

Additionally, we probably want to check that these objects are just copies, in that if you change them, they don't change the one in storage. 

After changing one, you have to store it again.

Reply:
Oh, and we probably want to be mean... and add maybe 1024 instruments. And we want to check that the order is kept.

Also, what want to check that setting an existing key retains the order. So, setting key "test_instrument1" to a new instrument should still keep it as the first.